pennylane/ops/op_math/linear_combination.py

```diff
@@ -321,58 +321,15 @@ def simplify(self, cutoff=1.0e-12):
         return LinearCombination(coeffs, ops, _pauli_rep=pr)
 
-    def _obs_data(self):
-        r"""Extracts the data from a ``LinearCombination`` and serializes it in an order-independent fashion.
-
-        This allows for comparison between ``LinearCombination``s that are equivalent, but are defined with terms and tensors
-        expressed in different orders. For example, `qml.X(0) @ qml.Z(1)` and
-        `qml.Z(1) @ qml.X(0)` are equivalent observables with different orderings.
-
-        .. Note::
-
-            In order to store the data from each term of the ``LinearCombination`` in an order-independent serialization,
-            we make use of sets. Note that all data contained within each term must be immutable, hence the use of
-            strings and frozensets.
-
-        **Example**
-
-        >>> H = qml.ops.LinearCombination([1, 1], [qml.X(0) @ qml.X(1), qml.Z(0)])
-        >>> print(H._obs_data())
-        {(1, frozenset({('Prod', <Wires = [0, 1]>, ())})),
-         (1, frozenset({('PauliZ', <Wires = [0]>, ())}))}
-        """
-        data = set()
-
-        coeffs_arr = qml.math.toarray(self.coeffs)
-        for co, op in zip(coeffs_arr, self.ops):
-            obs = op.non_identity_obs if isinstance(op, Tensor) else [op]
-            tensor = []
-            for ob in obs:
-                parameters = tuple(
-                    str(param) for param in ob.parameters
-                )  # Converts params into immutable type
-                if isinstance(ob, qml.GellMann):
-                    parameters += (ob.hyperparameters["index"],)
-                tensor.append((ob.name, ob.wires, parameters))
-            data.add((co, frozenset(tensor)))
-
-        return data
-
     def compare(self, other):
-        r"""Determines whether the operator is equivalent to another.
+        r"""Determines mathematical equivalence between operators
 
-        Currently only supported for :class:`~LinearCombination`, :class:`~.Observable`, or :class:`~.Tensor`.
-        LinearCombinations/observables are equivalent if they represent the same operator
-        (their matrix representations are equal), and they are defined on the same wires.
+        ``LinearCombination`` and other operators are equivalent if they mathematically represent the same operator
+        (their matrix representations are equal), acting on the same wires.
 
         .. Warning::
 
-            The compare method does **not** check if the matrix representation
-            of a :class:`~.Hermitian` observable is equal to an equivalent
-            observable expressed in terms of Pauli matrices, or as a
-            linear combination of Hermitians.
-            To do so would require the matrix form of LinearCombinations and Tensors
-            be calculated, which would drastically increase runtime.
+            While this method does not compute explicit matrices, it is still relatively expensive as it involves simplification of both operators.
```

Comment:
We can't try to use the Pauli representations, if available?

Reply:
That is what it is doing :)!

Reply:
Is comparing the Pauli representation expensive? I thought it was already in a simplified representation?

Reply:
Ah right now I get your point. Maybe I should say this _can_ be relatively expensive (in the cases where there is no pauli_rep) or perhaps leave out this comment about cost of this all together? Do you prefer a little detail or keeping it simple?

Reply:
Thanks! Maybe a sentence or two more detail can't hurt.